Writerside/topics/Quickstart-Up-And-Running.md

```diff
@@ -0,0 +1,585 @@
+
+### Configure Terraform
+
+The Terraform configuration defaults to using an S3 bucket to store it's state, this must be updated to an S3 bucket on your account. You may create this S3 bucket in the AWS Console.
+
+Open `terraform_config.tf` and modify the following portion to match your S3 bucket, or remove it to store state locally.
+
+```
+  backend "s3" {
+      bucket  = "catena-terraform-state"
+      key     = "updated-infra/terraform.tfstte"
+      region  = "us-east-1"
+      profile = "catena_admin"
+  }
+```
+
+Be sure to replace `bucket` with your bucket's name, `region` with your region, and `profile` with your local [AWS Profile](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/configure/index.html).
+
+> If you do not need to back up or share your terraform state, you can remove the "backend" block and terraform will store it's state locally.
+
+Next, we need to set some specific-to-you configurations.
+
+Create a `vars.tfvars` file similar to the following:
+
+```
+aws_profile="catena_admin"                  # Your AWS Profile
+aws_region="us-east-1"                      # The region you are working in
+ec2_ami="ami-0c7217cdde317cfec"             # AWS Ubuntu 22.04 LTS AMI for your region
+ec2_instance_size="t2.small"                # Instance Size for the machine running the catena backend
+ssh_private_key_path="~/.ssh/id_rsa"        # Your private key
+ssh_public_key_path="~/.ssh/id_rsa.pub"     # Your public key
+route_53_hosted_zone_name="catenatools.com" # The name of the hosted zone in route53
+letsencrypt_email="example@example.com"     # The email address LetsEncrypt will send cert expiry notifications to
+```
+
+### Provision the Infrastructure
+
+```bash
+terraform init                        # Initialize the terraform backend and install modules
+terraform workspace switch -or-create dev
+terraform plan -var-file="vars.tfvar" # Preview the steps terraform will take
+terraform apply -var-file="vars.tfvar"
+```
+
+After the apply step runs, you should see some output like this:
+
+```
+add_dokku_remote_command = "git remote add dokku dokku@dev.catenatools.com:platform"
+ec2_instance_id = "i-0f47e887113bf882b"
+ec2_instance_ssh_command = "aws ec2-instance-connect ssh --os-user ubuntu --instance-id i-0f47e887113bf882b --profile catena"
+ec2_ip = "100.28.26.117"
+powershell_deploy_command = "$env:GIT_SSH_COMMAND='ssh -i ~/.ssh/id_rsa -o IdentitiesOnly=yes'; git push dokku main"
+unix_deploy_command = "GIT_SSH_COMMAND='ssh -i ~/.ssh/id_rsa -o IdentitiesOnly=yes' git push dokku main"
+windows_deploy_command = "set \"GIT_SSH_COMMAND=ssh -i ~/.ssh/id_rsa -o IdentitiesOnly=yes\" && git push dokku main"
+catena_url = "https://platform.dev.catenatools.com"
+```
+
+These outputs will be used to deploy the platform to your newly configured instance.
```

Comment:
It might be worth stressing that these values should be saved somewhere for the rest of the setup or how to retrieve them. It's easy to lose track of things like this while following a guide for the first time. (Maybe we want to do something like write these out with `local_file` in the infrastructure?)

Reply:
This is a great idea! Maybe even create a source-able file?